rs/src/parse/plan.rs

```diff
@@ -62,15 +62,109 @@ fn parse_plan_rel(x: &substrait::PlanRel, y: &mut context::Context) -> diagnosti
     Ok(())
 }
 
-/// Toplevel parse function for a plan.
-pub fn parse_plan(x: &substrait::Plan, y: &mut context::Context) -> diagnostic::Result<()> {
+/// Parse a git hash string.
+fn parse_git_hash(x: &String, y: &mut context::Context) -> diagnostic::Result<()> {
+    if !x.is_empty() {
+        static GIT_HASH_RE: once_cell::sync::Lazy<regex::Regex> =
+            once_cell::sync::Lazy::new(|| regex::Regex::new("[0-9a-f]{40}").unwrap());
+        if !GIT_HASH_RE.is_match(x) {
```

Comment:
Meh. I don't need the actual value, I doubt your suggestion will check that the string is lowercase, and it seems a bit excessive to pull in a new dependency for something like this...

Reply:
Oh I missed that it needs to be lowercase. I would always prefer a dependency over writing regular expressions for parsing common things, but that's mainly for maintenance reasons. This regex isn't complicated at all, but it's just something I try to avoid. But I agree that in this case it's a bit excessive.